warehouse/query-core/src/main/java/datawave/query/iterator/ParentQueryIterator.java

```diff
@@ -59,11 +61,17 @@ public EventDataQueryFilter getEvaluationFilter() {
     @Override
     public Iterator<Entry<Key,Document>> mapDocument(SortedKeyValueIterator<Key,Value> deepSourceCopy, Iterator<Entry<Key,Document>> documents,
                     CompositeMetadata compositeMetadata) {
-        Iterator<Tuple2<Key,Document>> parentDocuments = Iterators.transform(documents, new GetParentDocument(
+        
         // no evaluation filter here as this is post evaluation
-                        new KeyToDocumentData(deepSourceCopy, this.myEnvironment, this.documentOptions, this.equality, null, this.includeHierarchyFields,
-                                        this.includeHierarchyFields), new Aggregation(this.getTimeFilter(), this.typeMetadataWithNonIndexed, compositeMetadata,
-                                        this.isIncludeGroupingContext(), this.includeRecordId, this.parentDisableIndexOnlyDocuments, null)));
+        
+        Aggregation aggregation = new Aggregation(this.getTimeFilter(), this.typeMetadataWithNonIndexed, compositeMetadata, this.isIncludeGroupingContext(),
+                        this.includeRecordId, this.parentDisableIndexOnlyDocuments, null);
+        
+        KeyToDocumentData k2d = new KeyToDocumentData(deepSourceCopy, this.myEnvironment, this.documentOptions, this.equality, null,
+                        this.includeHierarchyFields, this.includeHierarchyFields);
+        k2d.withScanRangeProvider(getScanRangeProvider());
```

Comment:
Why didn't AncestorQueryIterator require something similar?

Reply:
AncestoryQueryIterator overrides `getScanRangeProvider()` which hooks cleanly into the two places that the QueryIterator constructs a `KeyToDocumentData` object. The only other place in the code that constructs a `KeyToDocumentData` objct is the ParentQueryIterator. So this additional instance of a new KeyToDocumentData requires the scan range provider to be set.